services/user_service.py

```diff
@@ -35,9 +37,37 @@ def update_user(self, email: str, updated_data: dict) -> Optional[User]:
 
     def get_user_details(self, email: str) -> Optional[User]:
         return self.user_repository.get_user_by_email(email)
-
+    
     def remove_user(self, email: str):
         user = self.user_repository.get_user_by_email(email)
         if user is None:
             raise ValueError("User does not exist.")
-        return self.user_repository.delete_user(email)
\ No newline at end of file
+        return self.user_repository.delete_user(email)
+
+    def generate_profile_picture_path(self, email:str, image: UploadFile ):
+        user = self.user_repository.get_user_by_email(email)
+        if user is None:
+            raise ValueError("User does not exist.")
+        file_path = f"uploads/profile_pictures/{email}/{image.filename}" 
+        os.makedirs(os.path.dirname(file_path), exist_ok=True)
+        with open(file_path, "wb") as buffer:
+            shutil.copyfileobj(image.file, buffer)
```

Comment:
For this one, there is a scenario where, the file updating might fail after the file is saved, and in this case to maintain the atomicity, you should roll back the file creation to avoid the directory to retain unused files.